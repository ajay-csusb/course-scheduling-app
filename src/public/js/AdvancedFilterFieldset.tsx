import React from 'react';
import { Collapse, Label } from '@blueprintjs/core';
import { SelectListCourseAttr } from './SelectListCourseAttr';
import { SelectListSessionCode } from './SelectListSessionCode';
import { SelectListGeClassesAttributes } from './SelectListGeClassesAttributes';
import { IClassSearchFormProps } from './ClassSearchForm';

interface IAdvancedFilterFieldsetState {
  isOpen?: boolean;
}

export class AdvancedFilterFieldset extends React.Component<
    IClassSearchFormProps, IAdvancedFilterFieldsetState> {
  constructor(props: IClassSearchFormProps) {
    super(props);
    this.state = {
      isOpen: false,
    };
    this.handleClick = this.handleClick.bind(this);
  }

  public render() {
    const courseAttr = this.getSelectListCourseAttrComponent();
    const classNumber = this.getClassNumberComponent();
    const sessionCode = this.getSelectListSessionCodeComponent();
    const geClassesAttribute = this.getSelectListGeClassesAttributesComponent();
    const arrow = (this.state.isOpen) ? <i className="fal fa-chevron-up"/> : <i className="fal fa-chevron-down"/>;
    return (
      <React.Fragment>
        <a id="additional-filters" onClick={this.handleClick}>
          {this.state.isOpen ? 'Hide' : ''} Additional Search Criteria {arrow}
        </a>
        <Collapse isOpen={this.state.isOpen} >
          <div className="row">
            <div className="col-6 col-md-4">
              <div className="form-item">{geClassesAttribute}</div>
            </div>
            <div className="col-6 col-md-4">
              <div className="form-item">{courseAttr}</div>
            </div>
            <div className="col-6 col-md-4">
              <div className="form-item">{classNumber}</div>
            </div>
            <div className="col-6 col-md-4">
              <div className="form-item">{sessionCode}</div>
            </div>
          </div>
        </Collapse>
      </React.Fragment>
    );
  }

  private handleClick = () => {
    this.setState({ isOpen: !this.state.isOpen });
  }

  private getSelectListCourseAttrComponent(): JSX.Element {
    return (
      <SelectListCourseAttr
        courseAttr={this.props.courseAttr}
        onChangeOfCourseAttr={this.props.onChangeOfCourseAttr}
      />
    );
  }

  private getClassNumberComponent(): JSX.Element {
    return (
      <React.Fragment>
        <Label htmlFor="class-number">Class Number</Label>
        <input
          id="class-number"
          className="class-number"
          type="text"
          placeholder="E.g. 60256"
          dir="auto"
          onChange={this.props.onChangeOfClassNo}
          value={this.props.classNo}
          onKeyDown={this.props.onKeyDown}
        />
      </React.Fragment>
    );
  }

  private getSelectListSessionCodeComponent(): JSX.Element {
    return (
      <SelectListSessionCode
        sessionCode={this.props.sessionCode}
        onChangeOfSessionCode={this.props.onChangeOfSessionCode}
      />
    );
  }

  private getSelectListGeClassesAttributesComponent(): JSX.Element {
    return (
      <SelectListGeClassesAttributes
        geClassesAttributes={this.props.geClassesAttributes}
        geClassesAttribute={this.props.geClassesAttribute}
        onChangeOfGeClassesAttribute={this.props.onChangeOfGeClassesAttribute}
      />
    );
  }

}
