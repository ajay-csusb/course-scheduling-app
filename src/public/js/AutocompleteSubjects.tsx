import { IItemRendererProps } from '@blueprintjs/select';
import React from 'react';
import { MenuItem } from '@blueprintjs/core';
import { ISubject } from './Subject';

export function filterSubjects(query: string, subject: ISubject): boolean {
  const parsedQuery = query.toLowerCase().trim();
  if (subject.name.toLowerCase().trim().indexOf(parsedQuery) >= 0) {
    return true;
  }
  if (subject.abbr.toLowerCase().trim().indexOf(parsedQuery) >= 0) {
    return true;
  }
  return false;
}

function renderSubjects(subject: ISubject, classRenderer: IItemRendererProps): JSX.Element {
  return (
    <MenuItem
      active={false}
      disabled={false}
      label={subject.abbr}
      key={subject.abbr}
      onClick={classRenderer.handleClick}
      text={subject.name}
      className="search-autocomplete-items"
    />
  );
}

export function renderSubject(subject: ISubject): string {
  return `${subject.name.trim()} (${subject.abbr.trim()})`;
}

export const autocompleteSubjectsProps = {
  itemRenderer: renderSubjects,
  itemPredicate: filterSubjects,
  inputValueRenderer: renderSubject,
};
