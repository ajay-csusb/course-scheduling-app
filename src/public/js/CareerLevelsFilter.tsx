import { IClass, ICareerLevels } from './Class';

export function filter(classes: IClass[], careerLevelsOptions: ICareerLevels): IClass[] {
  const result: IClass[] = [];

  if (!atLeastOneOptionSelected(careerLevelsOptions)) {
    return classes;
  }

  classes.forEach(_class => {
    if (careerLevelsOptions.ugrd && _class.degreeType.toLowerCase() === 'ugrd') {
      result.push(_class);
    }
    if (careerLevelsOptions.pbac && _class.degreeType.toLowerCase() === 'pbac') {
      result.push(_class);
    }
    if (careerLevelsOptions.exed && _class.degreeType.toLowerCase() === 'exed') {
      result.push(_class);
    }
  });

  return result;
}

function atLeastOneOptionSelected(careerLevelsOptions: ICareerLevels) {
  return careerLevelsOptions.ugrd || careerLevelsOptions.pbac || careerLevelsOptions.exed;
}
