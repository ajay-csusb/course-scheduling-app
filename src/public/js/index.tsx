import * as React from 'react';
import * as ReactDOM from 'react-dom';
import { ClassSearchContainer } from './ClassSearchContainer';
import { Utils } from './Utils';

Utils.loadExternalTools();
if (Utils.isProd()) {
  const ErrorBoundary = Utils.loadBugsnagComponent();
  ReactDOM.render(<ErrorBoundary><ClassSearchContainer /></ErrorBoundary>,
    document.querySelector('#class-search-form-wrapper'));
} else {
  ReactDOM.render(<ClassSearchContainer />, document.querySelector('#class-search-form-wrapper'));
}
