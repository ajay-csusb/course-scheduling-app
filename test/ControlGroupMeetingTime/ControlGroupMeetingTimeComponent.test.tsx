import { mount } from 'enzyme';
import * as React from 'react';
import { ClassSearchContainer } from '../../src/public/js/ClassSearchContainer';
import { TestUtils } from '../TestUtils';
// tslint:disable:max-line-length
describe('Meeting time component', () => {

  beforeAll(() => {
    TestUtils.ajax();
  });

  describe('Given a class search form', () => {
    test('meeting time component snapshot', () => {
      const classSearchContainerWrapper = mount(<ClassSearchContainer />);
      const meetingTimeWrapper = classSearchContainerWrapper.find('.meeting-time');
      expect(meetingTimeWrapper).toMatchSnapshot();
    });
  });
});
