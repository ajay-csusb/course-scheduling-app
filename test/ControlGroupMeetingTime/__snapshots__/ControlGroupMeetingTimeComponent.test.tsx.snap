// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Meeting time component Given a class search form meeting time component snapshot 1`] = `
Array [
  <Blueprint3.ControlGroup
    className="meeting-time"
  >
    <div
      className="bp3-control-group meeting-time"
    >
      <Component>
        <label
          className="bp3-label"
        >
          Meeting Time
          <Component>
            <label
              className="bp3-label"
            >
              Start Time
              <Blueprint3.TimePicker
                className="start-time"
                disabled={false}
                maxTime={1900-01-01T07:59:59.999Z}
                minTime={1899-12-31T08:00:00.000Z}
                onChange={[Function]}
                precision="minute"
                selectAllOnFocus={true}
                showArrowButtons={true}
                useAmPm={true}
                value={1899-01-01T08:00:00.000Z}
              >
                <div
                  className="bp3-timepicker start-time"
                >
                  <div
                    className="bp3-timepicker-arrow-row"
                  >
                    <span
                      className="bp3-timepicker-arrow-button bp3-timepicker-hour"
                      onClick={[Function]}
                    >
                      <Blueprint3.Icon
                        icon="chevron-up"
                      >
                        <span
                          className="bp3-icon bp3-icon-chevron-up"
                          icon="chevron-up"
                        >
                          <svg
                            data-icon="chevron-up"
                            height={16}
                            viewBox="0 0 16 16"
                            width={16}
                          >
                            <desc>
                              chevron-up
                            </desc>
                            <path
                              d="M12.71 9.29l-4-4C8.53 5.11 8.28 5 8 5s-.53.11-.71.29l-4 4a1.003 1.003 0 0 0 1.42 1.42L8 7.41l3.29 3.29c.18.19.43.3.71.3a1.003 1.003 0 0 0 .71-1.71z"
                              fillRule="evenodd"
                              key="0"
                            />
                          </svg>
                        </span>
                      </Blueprint3.Icon>
                    </span>
                    <span
                      className="bp3-timepicker-arrow-button bp3-timepicker-minute"
                      onClick={[Function]}
                    >
                      <Blueprint3.Icon
                        icon="chevron-up"
                      >
                        <span
                          className="bp3-icon bp3-icon-chevron-up"
                          icon="chevron-up"
                        >
                          <svg
                            data-icon="chevron-up"
                            height={16}
                            viewBox="0 0 16 16"
                            width={16}
                          >
                            <desc>
                              chevron-up
                            </desc>
                            <path
                              d="M12.71 9.29l-4-4C8.53 5.11 8.28 5 8 5s-.53.11-.71.29l-4 4a1.003 1.003 0 0 0 1.42 1.42L8 7.41l3.29 3.29c.18.19.43.3.71.3a1.003 1.003 0 0 0 .71-1.71z"
                              fillRule="evenodd"
                              key="0"
                            />
                          </svg>
                        </span>
                      </Blueprint3.Icon>
                    </span>
                  </div>
                  <div
                    className="bp3-timepicker-input-row"
                  >
                    <input
                      className="bp3-timepicker-input bp3-timepicker-hour"
                      disabled={false}
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      value="12"
                    />
                    <span
                      className="bp3-timepicker-divider-text"
                    >
                      :
                    </span>
                    <input
                      className="bp3-timepicker-input bp3-timepicker-minute"
                      disabled={false}
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      value="00"
                    />
                  </div>
                  <HTMLSelect
                    className="bp3-timepicker-ampm-select"
                    disabled={false}
                    onChange={[Function]}
                    value="am"
                  >
                    <div
                      className="bp3-html-select bp3-timepicker-ampm-select"
                    >
                      <select
                        disabled={false}
                        multiple={false}
                        onChange={[Function]}
                        value="am"
                      >
                        <option
                          value="am"
                        >
                          AM
                        </option>
                        <option
                          value="pm"
                        >
                          PM
                        </option>
                      </select>
                      <Blueprint3.Icon
                        icon="double-caret-vertical"
                      >
                        <span
                          className="bp3-icon bp3-icon-double-caret-vertical"
                          icon="double-caret-vertical"
                        >
                          <svg
                            data-icon="double-caret-vertical"
                            height={16}
                            viewBox="0 0 16 16"
                            width={16}
                          >
                            <desc>
                              double-caret-vertical
                            </desc>
                            <path
                              d="M5 7h6a1.003 1.003 0 0 0 .71-1.71l-3-3C8.53 2.11 8.28 2 8 2s-.53.11-.71.29l-3 3A1.003 1.003 0 0 0 5 7zm6 2H5a1.003 1.003 0 0 0-.71 1.71l3 3c.18.18.43.29.71.29s.53-.11.71-.29l3-3A1.003 1.003 0 0 0 11 9z"
                              fillRule="evenodd"
                              key="0"
                            />
                          </svg>
                        </span>
                      </Blueprint3.Icon>
                    </div>
                  </HTMLSelect>
                  <div
                    className="bp3-timepicker-arrow-row"
                  >
                    <span
                      className="bp3-timepicker-arrow-button bp3-timepicker-hour"
                      onClick={[Function]}
                    >
                      <Blueprint3.Icon
                        icon="chevron-down"
                      >
                        <span
                          className="bp3-icon bp3-icon-chevron-down"
                          icon="chevron-down"
                        >
                          <svg
                            data-icon="chevron-down"
                            height={16}
                            viewBox="0 0 16 16"
                            width={16}
                          >
                            <desc>
                              chevron-down
                            </desc>
                            <path
                              d="M12 5c-.28 0-.53.11-.71.29L8 8.59l-3.29-3.3a1.003 1.003 0 0 0-1.42 1.42l4 4c.18.18.43.29.71.29s.53-.11.71-.29l4-4A1.003 1.003 0 0 0 12 5z"
                              fillRule="evenodd"
                              key="0"
                            />
                          </svg>
                        </span>
                      </Blueprint3.Icon>
                    </span>
                    <span
                      className="bp3-timepicker-arrow-button bp3-timepicker-minute"
                      onClick={[Function]}
                    >
                      <Blueprint3.Icon
                        icon="chevron-down"
                      >
                        <span
                          className="bp3-icon bp3-icon-chevron-down"
                          icon="chevron-down"
                        >
                          <svg
                            data-icon="chevron-down"
                            height={16}
                            viewBox="0 0 16 16"
                            width={16}
                          >
                            <desc>
                              chevron-down
                            </desc>
                            <path
                              d="M12 5c-.28 0-.53.11-.71.29L8 8.59l-3.29-3.3a1.003 1.003 0 0 0-1.42 1.42l4 4c.18.18.43.29.71.29s.53-.11.71-.29l4-4A1.003 1.003 0 0 0 12 5z"
                              fillRule="evenodd"
                              key="0"
                            />
                          </svg>
                        </span>
                      </Blueprint3.Icon>
                    </span>
                  </div>
                </div>
              </Blueprint3.TimePicker>
            </label>
          </Component>
          <Component>
            <label
              className="bp3-label"
            >
              End Time
              <Blueprint3.TimePicker
                className="end-time"
                disabled={false}
                maxTime={1900-01-01T07:59:59.999Z}
                minTime={1899-12-31T08:00:00.000Z}
                onChange={[Function]}
                precision="minute"
                selectAllOnFocus={true}
                showArrowButtons={true}
                useAmPm={true}
                value={1899-01-02T07:00:00.000Z}
              >
                <div
                  className="bp3-timepicker end-time"
                >
                  <div
                    className="bp3-timepicker-arrow-row"
                  >
                    <span
                      className="bp3-timepicker-arrow-button bp3-timepicker-hour"
                      onClick={[Function]}
                    >
                      <Blueprint3.Icon
                        icon="chevron-up"
                      >
                        <span
                          className="bp3-icon bp3-icon-chevron-up"
                          icon="chevron-up"
                        >
                          <svg
                            data-icon="chevron-up"
                            height={16}
                            viewBox="0 0 16 16"
                            width={16}
                          >
                            <desc>
                              chevron-up
                            </desc>
                            <path
                              d="M12.71 9.29l-4-4C8.53 5.11 8.28 5 8 5s-.53.11-.71.29l-4 4a1.003 1.003 0 0 0 1.42 1.42L8 7.41l3.29 3.29c.18.19.43.3.71.3a1.003 1.003 0 0 0 .71-1.71z"
                              fillRule="evenodd"
                              key="0"
                            />
                          </svg>
                        </span>
                      </Blueprint3.Icon>
                    </span>
                    <span
                      className="bp3-timepicker-arrow-button bp3-timepicker-minute"
                      onClick={[Function]}
                    >
                      <Blueprint3.Icon
                        icon="chevron-up"
                      >
                        <span
                          className="bp3-icon bp3-icon-chevron-up"
                          icon="chevron-up"
                        >
                          <svg
                            data-icon="chevron-up"
                            height={16}
                            viewBox="0 0 16 16"
                            width={16}
                          >
                            <desc>
                              chevron-up
                            </desc>
                            <path
                              d="M12.71 9.29l-4-4C8.53 5.11 8.28 5 8 5s-.53.11-.71.29l-4 4a1.003 1.003 0 0 0 1.42 1.42L8 7.41l3.29 3.29c.18.19.43.3.71.3a1.003 1.003 0 0 0 .71-1.71z"
                              fillRule="evenodd"
                              key="0"
                            />
                          </svg>
                        </span>
                      </Blueprint3.Icon>
                    </span>
                  </div>
                  <div
                    className="bp3-timepicker-input-row"
                  >
                    <input
                      className="bp3-timepicker-input bp3-timepicker-hour"
                      disabled={false}
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      value="11"
                    />
                    <span
                      className="bp3-timepicker-divider-text"
                    >
                      :
                    </span>
                    <input
                      className="bp3-timepicker-input bp3-timepicker-minute"
                      disabled={false}
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      value="00"
                    />
                  </div>
                  <HTMLSelect
                    className="bp3-timepicker-ampm-select"
                    disabled={false}
                    onChange={[Function]}
                    value="pm"
                  >
                    <div
                      className="bp3-html-select bp3-timepicker-ampm-select"
                    >
                      <select
                        disabled={false}
                        multiple={false}
                        onChange={[Function]}
                        value="pm"
                      >
                        <option
                          value="am"
                        >
                          AM
                        </option>
                        <option
                          value="pm"
                        >
                          PM
                        </option>
                      </select>
                      <Blueprint3.Icon
                        icon="double-caret-vertical"
                      >
                        <span
                          className="bp3-icon bp3-icon-double-caret-vertical"
                          icon="double-caret-vertical"
                        >
                          <svg
                            data-icon="double-caret-vertical"
                            height={16}
                            viewBox="0 0 16 16"
                            width={16}
                          >
                            <desc>
                              double-caret-vertical
                            </desc>
                            <path
                              d="M5 7h6a1.003 1.003 0 0 0 .71-1.71l-3-3C8.53 2.11 8.28 2 8 2s-.53.11-.71.29l-3 3A1.003 1.003 0 0 0 5 7zm6 2H5a1.003 1.003 0 0 0-.71 1.71l3 3c.18.18.43.29.71.29s.53-.11.71-.29l3-3A1.003 1.003 0 0 0 11 9z"
                              fillRule="evenodd"
                              key="0"
                            />
                          </svg>
                        </span>
                      </Blueprint3.Icon>
                    </div>
                  </HTMLSelect>
                  <div
                    className="bp3-timepicker-arrow-row"
                  >
                    <span
                      className="bp3-timepicker-arrow-button bp3-timepicker-hour"
                      onClick={[Function]}
                    >
                      <Blueprint3.Icon
                        icon="chevron-down"
                      >
                        <span
                          className="bp3-icon bp3-icon-chevron-down"
                          icon="chevron-down"
                        >
                          <svg
                            data-icon="chevron-down"
                            height={16}
                            viewBox="0 0 16 16"
                            width={16}
                          >
                            <desc>
                              chevron-down
                            </desc>
                            <path
                              d="M12 5c-.28 0-.53.11-.71.29L8 8.59l-3.29-3.3a1.003 1.003 0 0 0-1.42 1.42l4 4c.18.18.43.29.71.29s.53-.11.71-.29l4-4A1.003 1.003 0 0 0 12 5z"
                              fillRule="evenodd"
                              key="0"
                            />
                          </svg>
                        </span>
                      </Blueprint3.Icon>
                    </span>
                    <span
                      className="bp3-timepicker-arrow-button bp3-timepicker-minute"
                      onClick={[Function]}
                    >
                      <Blueprint3.Icon
                        icon="chevron-down"
                      >
                        <span
                          className="bp3-icon bp3-icon-chevron-down"
                          icon="chevron-down"
                        >
                          <svg
                            data-icon="chevron-down"
                            height={16}
                            viewBox="0 0 16 16"
                            width={16}
                          >
                            <desc>
                              chevron-down
                            </desc>
                            <path
                              d="M12 5c-.28 0-.53.11-.71.29L8 8.59l-3.29-3.3a1.003 1.003 0 0 0-1.42 1.42l4 4c.18.18.43.29.71.29s.53-.11.71-.29l4-4A1.003 1.003 0 0 0 12 5z"
                              fillRule="evenodd"
                              key="0"
                            />
                          </svg>
                        </span>
                      </Blueprint3.Icon>
                    </span>
                  </div>
                </div>
              </Blueprint3.TimePicker>
            </label>
          </Component>
        </label>
      </Component>
    </div>
  </Blueprint3.ControlGroup>,
  <div
    className="bp3-control-group meeting-time"
  >
    <Component>
      <label
        className="bp3-label"
      >
        Meeting Time
        <Component>
          <label
            className="bp3-label"
          >
            Start Time
            <Blueprint3.TimePicker
              className="start-time"
              disabled={false}
              maxTime={1900-01-01T07:59:59.999Z}
              minTime={1899-12-31T08:00:00.000Z}
              onChange={[Function]}
              precision="minute"
              selectAllOnFocus={true}
              showArrowButtons={true}
              useAmPm={true}
              value={1899-01-01T08:00:00.000Z}
            >
              <div
                className="bp3-timepicker start-time"
              >
                <div
                  className="bp3-timepicker-arrow-row"
                >
                  <span
                    className="bp3-timepicker-arrow-button bp3-timepicker-hour"
                    onClick={[Function]}
                  >
                    <Blueprint3.Icon
                      icon="chevron-up"
                    >
                      <span
                        className="bp3-icon bp3-icon-chevron-up"
                        icon="chevron-up"
                      >
                        <svg
                          data-icon="chevron-up"
                          height={16}
                          viewBox="0 0 16 16"
                          width={16}
                        >
                          <desc>
                            chevron-up
                          </desc>
                          <path
                            d="M12.71 9.29l-4-4C8.53 5.11 8.28 5 8 5s-.53.11-.71.29l-4 4a1.003 1.003 0 0 0 1.42 1.42L8 7.41l3.29 3.29c.18.19.43.3.71.3a1.003 1.003 0 0 0 .71-1.71z"
                            fillRule="evenodd"
                            key="0"
                          />
                        </svg>
                      </span>
                    </Blueprint3.Icon>
                  </span>
                  <span
                    className="bp3-timepicker-arrow-button bp3-timepicker-minute"
                    onClick={[Function]}
                  >
                    <Blueprint3.Icon
                      icon="chevron-up"
                    >
                      <span
                        className="bp3-icon bp3-icon-chevron-up"
                        icon="chevron-up"
                      >
                        <svg
                          data-icon="chevron-up"
                          height={16}
                          viewBox="0 0 16 16"
                          width={16}
                        >
                          <desc>
                            chevron-up
                          </desc>
                          <path
                            d="M12.71 9.29l-4-4C8.53 5.11 8.28 5 8 5s-.53.11-.71.29l-4 4a1.003 1.003 0 0 0 1.42 1.42L8 7.41l3.29 3.29c.18.19.43.3.71.3a1.003 1.003 0 0 0 .71-1.71z"
                            fillRule="evenodd"
                            key="0"
                          />
                        </svg>
                      </span>
                    </Blueprint3.Icon>
                  </span>
                </div>
                <div
                  className="bp3-timepicker-input-row"
                >
                  <input
                    className="bp3-timepicker-input bp3-timepicker-hour"
                    disabled={false}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    value="12"
                  />
                  <span
                    className="bp3-timepicker-divider-text"
                  >
                    :
                  </span>
                  <input
                    className="bp3-timepicker-input bp3-timepicker-minute"
                    disabled={false}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    value="00"
                  />
                </div>
                <HTMLSelect
                  className="bp3-timepicker-ampm-select"
                  disabled={false}
                  onChange={[Function]}
                  value="am"
                >
                  <div
                    className="bp3-html-select bp3-timepicker-ampm-select"
                  >
                    <select
                      disabled={false}
                      multiple={false}
                      onChange={[Function]}
                      value="am"
                    >
                      <option
                        value="am"
                      >
                        AM
                      </option>
                      <option
                        value="pm"
                      >
                        PM
                      </option>
                    </select>
                    <Blueprint3.Icon
                      icon="double-caret-vertical"
                    >
                      <span
                        className="bp3-icon bp3-icon-double-caret-vertical"
                        icon="double-caret-vertical"
                      >
                        <svg
                          data-icon="double-caret-vertical"
                          height={16}
                          viewBox="0 0 16 16"
                          width={16}
                        >
                          <desc>
                            double-caret-vertical
                          </desc>
                          <path
                            d="M5 7h6a1.003 1.003 0 0 0 .71-1.71l-3-3C8.53 2.11 8.28 2 8 2s-.53.11-.71.29l-3 3A1.003 1.003 0 0 0 5 7zm6 2H5a1.003 1.003 0 0 0-.71 1.71l3 3c.18.18.43.29.71.29s.53-.11.71-.29l3-3A1.003 1.003 0 0 0 11 9z"
                            fillRule="evenodd"
                            key="0"
                          />
                        </svg>
                      </span>
                    </Blueprint3.Icon>
                  </div>
                </HTMLSelect>
                <div
                  className="bp3-timepicker-arrow-row"
                >
                  <span
                    className="bp3-timepicker-arrow-button bp3-timepicker-hour"
                    onClick={[Function]}
                  >
                    <Blueprint3.Icon
                      icon="chevron-down"
                    >
                      <span
                        className="bp3-icon bp3-icon-chevron-down"
                        icon="chevron-down"
                      >
                        <svg
                          data-icon="chevron-down"
                          height={16}
                          viewBox="0 0 16 16"
                          width={16}
                        >
                          <desc>
                            chevron-down
                          </desc>
                          <path
                            d="M12 5c-.28 0-.53.11-.71.29L8 8.59l-3.29-3.3a1.003 1.003 0 0 0-1.42 1.42l4 4c.18.18.43.29.71.29s.53-.11.71-.29l4-4A1.003 1.003 0 0 0 12 5z"
                            fillRule="evenodd"
                            key="0"
                          />
                        </svg>
                      </span>
                    </Blueprint3.Icon>
                  </span>
                  <span
                    className="bp3-timepicker-arrow-button bp3-timepicker-minute"
                    onClick={[Function]}
                  >
                    <Blueprint3.Icon
                      icon="chevron-down"
                    >
                      <span
                        className="bp3-icon bp3-icon-chevron-down"
                        icon="chevron-down"
                      >
                        <svg
                          data-icon="chevron-down"
                          height={16}
                          viewBox="0 0 16 16"
                          width={16}
                        >
                          <desc>
                            chevron-down
                          </desc>
                          <path
                            d="M12 5c-.28 0-.53.11-.71.29L8 8.59l-3.29-3.3a1.003 1.003 0 0 0-1.42 1.42l4 4c.18.18.43.29.71.29s.53-.11.71-.29l4-4A1.003 1.003 0 0 0 12 5z"
                            fillRule="evenodd"
                            key="0"
                          />
                        </svg>
                      </span>
                    </Blueprint3.Icon>
                  </span>
                </div>
              </div>
            </Blueprint3.TimePicker>
          </label>
        </Component>
        <Component>
          <label
            className="bp3-label"
          >
            End Time
            <Blueprint3.TimePicker
              className="end-time"
              disabled={false}
              maxTime={1900-01-01T07:59:59.999Z}
              minTime={1899-12-31T08:00:00.000Z}
              onChange={[Function]}
              precision="minute"
              selectAllOnFocus={true}
              showArrowButtons={true}
              useAmPm={true}
              value={1899-01-02T07:00:00.000Z}
            >
              <div
                className="bp3-timepicker end-time"
              >
                <div
                  className="bp3-timepicker-arrow-row"
                >
                  <span
                    className="bp3-timepicker-arrow-button bp3-timepicker-hour"
                    onClick={[Function]}
                  >
                    <Blueprint3.Icon
                      icon="chevron-up"
                    >
                      <span
                        className="bp3-icon bp3-icon-chevron-up"
                        icon="chevron-up"
                      >
                        <svg
                          data-icon="chevron-up"
                          height={16}
                          viewBox="0 0 16 16"
                          width={16}
                        >
                          <desc>
                            chevron-up
                          </desc>
                          <path
                            d="M12.71 9.29l-4-4C8.53 5.11 8.28 5 8 5s-.53.11-.71.29l-4 4a1.003 1.003 0 0 0 1.42 1.42L8 7.41l3.29 3.29c.18.19.43.3.71.3a1.003 1.003 0 0 0 .71-1.71z"
                            fillRule="evenodd"
                            key="0"
                          />
                        </svg>
                      </span>
                    </Blueprint3.Icon>
                  </span>
                  <span
                    className="bp3-timepicker-arrow-button bp3-timepicker-minute"
                    onClick={[Function]}
                  >
                    <Blueprint3.Icon
                      icon="chevron-up"
                    >
                      <span
                        className="bp3-icon bp3-icon-chevron-up"
                        icon="chevron-up"
                      >
                        <svg
                          data-icon="chevron-up"
                          height={16}
                          viewBox="0 0 16 16"
                          width={16}
                        >
                          <desc>
                            chevron-up
                          </desc>
                          <path
                            d="M12.71 9.29l-4-4C8.53 5.11 8.28 5 8 5s-.53.11-.71.29l-4 4a1.003 1.003 0 0 0 1.42 1.42L8 7.41l3.29 3.29c.18.19.43.3.71.3a1.003 1.003 0 0 0 .71-1.71z"
                            fillRule="evenodd"
                            key="0"
                          />
                        </svg>
                      </span>
                    </Blueprint3.Icon>
                  </span>
                </div>
                <div
                  className="bp3-timepicker-input-row"
                >
                  <input
                    className="bp3-timepicker-input bp3-timepicker-hour"
                    disabled={false}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    value="11"
                  />
                  <span
                    className="bp3-timepicker-divider-text"
                  >
                    :
                  </span>
                  <input
                    className="bp3-timepicker-input bp3-timepicker-minute"
                    disabled={false}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    value="00"
                  />
                </div>
                <HTMLSelect
                  className="bp3-timepicker-ampm-select"
                  disabled={false}
                  onChange={[Function]}
                  value="pm"
                >
                  <div
                    className="bp3-html-select bp3-timepicker-ampm-select"
                  >
                    <select
                      disabled={false}
                      multiple={false}
                      onChange={[Function]}
                      value="pm"
                    >
                      <option
                        value="am"
                      >
                        AM
                      </option>
                      <option
                        value="pm"
                      >
                        PM
                      </option>
                    </select>
                    <Blueprint3.Icon
                      icon="double-caret-vertical"
                    >
                      <span
                        className="bp3-icon bp3-icon-double-caret-vertical"
                        icon="double-caret-vertical"
                      >
                        <svg
                          data-icon="double-caret-vertical"
                          height={16}
                          viewBox="0 0 16 16"
                          width={16}
                        >
                          <desc>
                            double-caret-vertical
                          </desc>
                          <path
                            d="M5 7h6a1.003 1.003 0 0 0 .71-1.71l-3-3C8.53 2.11 8.28 2 8 2s-.53.11-.71.29l-3 3A1.003 1.003 0 0 0 5 7zm6 2H5a1.003 1.003 0 0 0-.71 1.71l3 3c.18.18.43.29.71.29s.53-.11.71-.29l3-3A1.003 1.003 0 0 0 11 9z"
                            fillRule="evenodd"
                            key="0"
                          />
                        </svg>
                      </span>
                    </Blueprint3.Icon>
                  </div>
                </HTMLSelect>
                <div
                  className="bp3-timepicker-arrow-row"
                >
                  <span
                    className="bp3-timepicker-arrow-button bp3-timepicker-hour"
                    onClick={[Function]}
                  >
                    <Blueprint3.Icon
                      icon="chevron-down"
                    >
                      <span
                        className="bp3-icon bp3-icon-chevron-down"
                        icon="chevron-down"
                      >
                        <svg
                          data-icon="chevron-down"
                          height={16}
                          viewBox="0 0 16 16"
                          width={16}
                        >
                          <desc>
                            chevron-down
                          </desc>
                          <path
                            d="M12 5c-.28 0-.53.11-.71.29L8 8.59l-3.29-3.3a1.003 1.003 0 0 0-1.42 1.42l4 4c.18.18.43.29.71.29s.53-.11.71-.29l4-4A1.003 1.003 0 0 0 12 5z"
                            fillRule="evenodd"
                            key="0"
                          />
                        </svg>
                      </span>
                    </Blueprint3.Icon>
                  </span>
                  <span
                    className="bp3-timepicker-arrow-button bp3-timepicker-minute"
                    onClick={[Function]}
                  >
                    <Blueprint3.Icon
                      icon="chevron-down"
                    >
                      <span
                        className="bp3-icon bp3-icon-chevron-down"
                        icon="chevron-down"
                      >
                        <svg
                          data-icon="chevron-down"
                          height={16}
                          viewBox="0 0 16 16"
                          width={16}
                        >
                          <desc>
                            chevron-down
                          </desc>
                          <path
                            d="M12 5c-.28 0-.53.11-.71.29L8 8.59l-3.29-3.3a1.003 1.003 0 0 0-1.42 1.42l4 4c.18.18.43.29.71.29s.53-.11.71-.29l4-4A1.003 1.003 0 0 0 12 5z"
                            fillRule="evenodd"
                            key="0"
                          />
                        </svg>
                      </span>
                    </Blueprint3.Icon>
                  </span>
                </div>
              </div>
            </Blueprint3.TimePicker>
          </label>
        </Component>
      </label>
    </Component>
  </div>,
]
`;
